python/pyarrow/__init__.py

```diff
@@ -108,3 +108,9 @@ def jemalloc_memory_pool():
 
 
 localfs = LocalFilesystem.get_instance()
+
+
+# ----------------------------------------------------------------------
+# 0.4.0 deprecations
+
+InMemoryOutputStream = BufferOutputStream
```

Comment:
Totally. I added deprecation warnings for the other recently-renamed classes and a test suite to check that they still work properly